modules/imgproc/include/visp3/imgproc/vpCircleHoughTransform.h

```diff
@@ -0,0 +1,825 @@
+/****************************************************************************
+ *
+ * ViSP, open source Visual Servoing Platform software.
+ * Copyright (C) 2005 - 2023 by Inria. All rights reserved.
+ *
+ * This software is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file LICENSE.txt at the root directory of this source
+ * distribution for additional information about the GNU GPL.
+ *
+ * For using ViSP with software that can not be combined with the GNU
+ * GPL, please contact Inria about acquiring a ViSP Professional
+ * Edition License.
+ *
+ * See https://visp.inria.fr for more information.
+ *
+ * This software was developed at:
+ * Inria Rennes - Bretagne Atlantique
+ * Campus Universitaire de Beaulieu
+ * 35042 Rennes Cedex
+ * France
+ *
+ * If you have questions regarding the use of this file, please contact
+ * Inria at visp@inria.fr
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+ * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ *****************************************************************************/
+
+#ifndef _vpCircleHoughTransform_h_
+#define _vpCircleHoughTransform_h_
+
+// System includes
+#include <utility>
+#include <vector>
+
+// ViSP includes
+#include <visp3/core/vpConfig.h>
+#include <visp3/core/vpImage.h>
+#include <visp3/core/vpImageDraw.h>
+#include <visp3/core/vpImagePoint.h>
+#include <visp3/core/vpMatrix.h>
+#include <visp3/core/vpRect.h>
+#include <visp3/imgproc/vpImageMedian.h>
+
+// 3rd parties inclue
+#ifdef VISP_HAVE_NLOHMANN_JSON
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
+#endif
+
+/**
+ * \brief Class that permits to detect 2D circles in a image using
+ * the gradient-based Circle Hough transform.
+ * Please find more information on the algorithm 
+ * [here](https://theailearner.com/tag/hough-gradient-method-opencv/)
+ * 
+ */
+class VISP_EXPORT vpCircleHoughTransform
+{
+public:
+  /**
+   * \brief Class that gather the algorithm parameters.
+   */
+  class CHTransformParameters
+  {
```

Comment:
The name of this class doesn't conform ViSP style. Should be something like `vpCircleHoughParameters`